### DISCUSSION TITLE
wrong display of negative sml values at main html page

### DISCUSSION THREAD

**Discussion by iot-sle** [arendst/Tasmota#15177] [Q&A] [answered]
I use sml interface to read meter values. Some of them are negative. They are displayed correctly when the output is created by the corresponding sml line. Count of values is high, so I dont't use the standard output for sml values. They are hidden by * in the definition like in this example:

`2,04FDC8FF8100uuUUuuUUs@10,*,V,U_L1,1 ;19 U L1`

and use my own output:

```
>W
so(7)
<style>th,td { width:200px; }</style>
<table>
<tr><th> </th><td><b>ABB</td><td><b>ESY</td></tr>
<tr><th>ID:<td>%sml[11]%</td><td>%sml[1]%</td></tr> 
<tr><th>E Bezug :</th><td>%2sml[13]% kWh</td><td>%2sml[2]% kWh</td></tr>
<tr><th>E Einsp.:</th><td>%2sml[14]% kWh</td><td>%2sml[3]% kWh</td></tr>
<tr><th>E Gesamt:</th><td>%2sml[12]% kWh</td><td>        </td></tr>
<tr><th>P     L1:</th><td>%2sml[16]%   W</td><td>%2sml[5]%  W</td></tr>
<tr><th>P     L2:</th><td>%2sml[17]%   W</td><td>%2sml[6]%  W</td></tr>
<tr><th>P     L3:</th><td>%2sml[18]%   W</td><td>%2sml[7]%  W</td></tr>
<tr><th>P Gesamt:</th><td>%2sml[15]%   W</td><td>%2sml[4]%  W</td></tr>
<tr><th>U     L1:</th><td>%2sml[19]%   V</td><td>%2sml[8]%  V</td></tr>
<tr><th>U     L2:</th><td>%2sml[20]%   V</td><td>%2sml[9]%  V</td></tr>
<tr><th>U     L3:</th><td>%2sml[21]%   V</td><td>%2sml[10]%  V</td></tr>
<tr><th>Frequenz:</th><td>%2sml[22]%  Hz</td>
</table>
```
As you can see in result:

![grafik](https://user-images.githubusercontent.com/74416643/159116723-66f4ad95-3e55-43d5-922b-45d4a307dd31.png)

the negative values are formated in a wrong way. Is it a problem inside Tasmota? Or is something wrong with my web code? I don't need the leading zero here.

**Comment by gemu2015** [comment, marked as answer]
use formatter e.g.:
%0.2sml[17]%

**Comment by iot-sle** [comment]
thank you. That's it. Working fine now.
